### ISSUE TITLE
Online/Continuous Learning Support

### ISSUE THREAD

**Issue by robme-l** [automl/TabPFN#58]
I am curious on whether TabPFN's performance has been tested in an online learning setting, where `batch_size = 1`. Checking out some papers found some other AutoML frameworks deteriorated in inference performance when concept drift set in. Would significant modifications need to be taken for TabPFN or given its fast training/inference would a simple retrain be feasible?

**Comment by noahho** [comment]
We never tested TabPFN specifically in this setting, given the fast retrain of TabPFN I think retraining on an updated testset is feasible if the labels are available.

All the best

**Comment by robme-l** [comment]
@noahho So after testing it out with a custom train/eval loop that used an expanding window (to mimic iteratively receiving new data in an online setting), I came across a fundamental roadblock: `TabPFN` naturally infers classes, and so unless I feed it a few data points that show all the available classes (or at least 2 different ones, which would require a larger sample size for imbalanced datasets), it would not train.

At this point this is out of the scope of what I was working, although I did create wrapper classes and adapted dataloaders, it seems that in order to make TabPFN truly online would require a design decision within the model itself. What I do see however is that if one already has a large enough dataset that observes all classes, TabPFN can overcome this barrier. Thought you might be interested in this feedback.

**Comment by noahho** [comment]
Thanks a lot for the feedback! Does it even make sense to make predictions before you have observed two classes? In my understanding, there wouldn't be anything to predict before and you could start making predictions when you observed two classes. Am I misunderstanding that?

**Comment by robme-l** [comment]
@noahho I didn't get a reply notification. The idea is that the model doesn't necessarily need to know which classes exist beforehand but rather learns them and their distribution through iterative training/prediction. I am basing most of this with my experience with RiverML's library. I was looking for ways to integrate AutoML features with that library since it's very versatile, well supported, however their automated aspects are still under development and previous projects like FLAML's ChaCha are grossly neglected.

Like I mentioned this can be overcome with a large enough dataset size where one observes all the existing classes beforehand. However that kind of defeats the purpose of online learning. I am still a fan of TabPFN and if you have thoughts as to how we could adapt a version of it for online learning I am quite willing to work on it since, as I mentioned, I think both projects are great! Closing this issue here since I think this would count more as a discussion.
